sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1058,9 +1064,16 @@ def test_valid_brute_metric_for_auto_algorithm():
     require_params = ['mahalanobis', 'wminkowski', 'seuclidean']
     for metric in VALID_METRICS['brute']:
         if metric != 'precomputed' and metric not in require_params:
-            nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
-                                            metric=metric).fit(X)
-            nn.kneighbors(X)
+            if metric != 'haversine':
+                nn = neighbors.NearestNeighbors(n_neighbors=3,
+                                                algorithm='auto',
+                                                metric=metric).fit(X)
+                nn.kneighbors(X)
+            else:
+                nn = neighbors.NearestNeighbors(n_neighbors=3,
+                                                algorithm='auto',
+                                                metric=metric).fit(X[:, :2])
+                nn.kneighbors(X[:, :2])
```

Comment:
Probably better to use the same approach as above with slicing X to first 2 columns to be consistent.